src/Sylius/Component/Variation/spec/Checker/VariantsParityCheckerSpec.php

```diff
@@ -0,0 +1,119 @@
+class VariantsParityCheckerSpec extends ObjectBehavior
+{
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Variation\Checker\VariantsParityChecker');
+    }
+
+    function it_implements_variants_parity_checker_interface()
+    {
+        $this->shouldImplement(VariantsParityCheckerInterface::class);
+    }
+
+    function it_matches_if_given_variable_contains_similar_variant(
+        VariableInterface $tShirt,
+        VariantInterface $checkMediumBlueTShirt,
+        VariantInterface $smallBlueTShirt,
+        VariantInterface $mediumBlueTShirt,
+        OptionValueInterface $blueColor,
+        OptionValueInterface $mediumSize,
+        OptionInterface $color,
+        OptionInterface $size
+    ) {
+        $tShirt->getVariants()->willReturn([$smallBlueTShirt, $mediumBlueTShirt]);
+        $tShirt->getOptions()->willReturn([$color, $size]);
+
+        $smallBlueTShirt->hasOption($blueColor)->willReturn(true);
+        $smallBlueTShirt->hasOption($mediumSize)->willReturn(false);
+        $mediumBlueTShirt->hasOption($blueColor)->willReturn(true);
+        $mediumBlueTShirt->hasOption($mediumSize)->willReturn(true);
+
+        $checkMediumBlueTShirt->getOptions()->willReturn([$blueColor, $mediumSize]);
+
+        $this->checkParity($checkMediumBlueTShirt, $tShirt)->shouldReturn(true);
+    }
+
+    function it_returns_false_if_variable_does_not_contain_variant_with_same_options_set(
+        VariableInterface $tShirt,
+        VariantInterface $checkLargeBlueTShirt,
+        VariantInterface $smallBlueTShirt,
+        VariantInterface $mediumBlueTShirt,
+        OptionValueInterface $blueColor,
+        OptionValueInterface $largeSize,
+        OptionInterface $color,
+        OptionInterface $size
+    ) {
+        $tShirt->getVariants()->willReturn([$smallBlueTShirt, $mediumBlueTShirt]);
+        $tShirt->getOptions()->willReturn([$color, $size]);
+
+        $smallBlueTShirt->hasOption($blueColor)->willReturn(true);
+        $smallBlueTShirt->hasOption($largeSize)->willReturn(false);
+        $mediumBlueTShirt->hasOption($blueColor)->willReturn(true);
+        $mediumBlueTShirt->hasOption($largeSize)->willReturn(false);
+
+        $checkLargeBlueTShirt->getOptions()->willReturn([$blueColor, $largeSize]);
+
+        $this->checkParity($checkLargeBlueTShirt, $tShirt)->shouldReturn(false);
+    }
+
+    function it_matches_if_given_variable_has_given_variant(
+        VariableInterface $tShirt,
+        VariantInterface $checkMediumBlueTShirt,
+        OptionValueInterface $blueColor,
+        OptionValueInterface $mediumSize,
+        OptionInterface $color,
+        OptionInterface $size
+    ) {
+        $tShirt->getVariants()->willReturn([$checkMediumBlueTShirt]);
+        $tShirt->getOptions()->willReturn([$color, $size]);
+
+        $checkMediumBlueTShirt->getOptions()->willReturn([$blueColor, $mediumSize]);
+
+        $this->checkParity($checkMediumBlueTShirt, $tShirt)->shouldReturn(false);
```

Comment:
When the variant is the same as already existing one on a product shouldn't we return `true`?